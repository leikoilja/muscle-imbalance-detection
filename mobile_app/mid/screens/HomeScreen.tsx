import * as React from "react";
import { StyleSheet } from "react-native";

import { Text, View } from "react-native";
import { connect } from "react-redux";
import { watchPersonData } from "../state/store";

const mapStateToProps = (state) => {
  return {
    personData: state.personData,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    watchPersonData: () => dispatch(watchPersonData()),
  };
};

class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.props.watchPersonData();
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>HELP</Text>
        <Text>{this.props.personData.firstName}</Text>
        <Text>{this.props.personData.lastName}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
