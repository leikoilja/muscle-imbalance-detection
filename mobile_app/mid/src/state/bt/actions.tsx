export enum BT_ACTION_TYPES {
  SAVE_DEVICE = "BT/SAVE_DEVICE",
  UPDATE_DEVICE_STATE = "BR/UPDATE_DEVICE_STATE",
}

export const saveBtDevice = (device) => ({
  type: BT_ACTION_TYPES.SAVE_DEVICE,
  device,
});

export const updateDeviceIsConnected = (isConnected) => ({
  type: BT_ACTION_TYPES.UPDATE_DEVICE_STATE,
  isConnected,
});
