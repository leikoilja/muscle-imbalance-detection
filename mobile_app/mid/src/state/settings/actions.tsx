export enum SETTINGS_ACTION_TYPES {
  BT_SAVE_DEVICE = "SETTINGS/BT/SAVE_DEVICE",
  BT_UPDATE_DEVICE_STATE = "SETTINGS/BT/UPDATE_DEVICE_STATE",
  THEME_CHANGE = "SETTINGS/THEME/CHANGE",
}

export const saveBtDevice = (device) => ({
  type: SETTINGS_ACTION_TYPES.BT_SAVE_DEVICE,
  device,
});

export const updateDeviceIsConnected = (isConnected) => ({
  type: SETTINGS_ACTION_TYPES.BT_UPDATE_DEVICE_STATE,
  isConnected,
});

export const changeTheme = (theme) => ({
  type: SETTINGS_ACTION_TYPES.THEME_CHANGE,
  theme,
});
