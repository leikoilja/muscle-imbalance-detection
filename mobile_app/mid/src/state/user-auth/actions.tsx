import { firebase } from "../../firebase/config";
import {
  LogoutFinishedAction,
  User,
  LoginStartAction,
  LoginFinishedAction,
  LoginErrorAction,
  LoginUserAction,
  LogoutUserAction,
  UserAuthAction,
  LogoutStartAction,
  LogoutErrorAction,
  RegistrationErrorAction,
  RegistrationStartAction,
  RegistrationFinishedAction,
  RegisterUserAction,
} from "../types";
import { Dispatch } from "redux";

export enum USER_AUTH_ACTION_TYPES {
  LOGIN_START = "USER_AUTH/LOGIN_START",
  LOGIN_FINISHED = "USER_AUTH/LOGIN_FINISHED",
  LOGIN_ERROR = "USER_AUTH/LOGIN_ERROR",
  LOGOUT_START = "USER_AUTH/LOGOUT_START",
  LOGOUT_FINISHED = "USER_AUTH/LOGOUT_FINISHED",
  LOGOUT_ERROR = "USER_AUTH/LOGOUT_ERROR",
  REGISTRATION_START = "USER_AUTH/REGISTRATION_START",
  REGISTRATION_FINISHED = "USER_AUTH/REGISTRATION_FINISHED",
  REGISTRATION_ERROR = "USER_AUTH/REGISTRATION_ERROR",
  SAVE_MEASUREMENT = "USER/SAVE_MEASUREMENT",
}

// Login
export const loginStart = (): LoginStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_START,
});

export const loginFinished = (user: User): LoginFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_FINISHED,
  user,
});

export const loginError = (error: string): LoginErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_ERROR,
  error,
});

// Logout
export const logoutStart = (): LogoutStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_START,
});

export const logoutFinished = (): LogoutFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_FINISHED,
});

export const logoutError = (error: string): LogoutErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_ERROR,
  error,
});

// Registration
export const registrationStart = (): RegistrationStartAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_START,
});

export const registrationFinished = (
  user: User
): RegistrationFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_FINISHED,
  user,
});

export const registrationError = (error: string): RegistrationErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_ERROR,
  error,
});

// User related
export const saveMeasurement = (measurementUid: string) => ({
  type: USER_AUTH_ACTION_TYPES.SAVE_MEASUREMENT,
  measurementUid,
});

//
// Action creators
//
export const loginUser = (
  email: string,
  password: string
): LoginUserAction => async (dispatch) => {
  dispatch(loginStart());
  let user_data = {};
  await firebase
    .auth()
    .signInWithEmailAndPassword(email, password)
    .then((response) => {
      let user = response.user;
      firebase
        .firestore()
        .collection("users")
        .doc(user.uid)
        .get()
        .then((doc) => {
          if (!doc.exists) {
            console.log("No documents");
          } else {
            user_data = doc.data();
          }
        })
        .catch((error) => {
          dispatch(loginError(error));
        })
        .finally(() => {
          user = { ...user_data, user };
          dispatch(loginFinished(user));
        });
    })
    .catch((error) => {
      dispatch(loginError(error));
    });
};

export const logoutUser = (): LogoutUserAction => async (
  dispatch: Dispatch<UserAuthAction>
) => {
  dispatch(logoutStart());
  try {
    await firebase.auth().signOut();
    dispatch(logoutFinished());
  } catch (error) {
    dispatch(logoutError(error));
  }
};

export const registerUser = (
  email: string,
  password: string,
  fullName: string
): RegisterUserAction => async (dispatch) => {
  dispatch(registrationStart());
  await firebase
    .auth()
    .createUserWithEmailAndPassword(email, password)
    .then((response) => {
      const user = response.user;
      const account = {
        fullName: fullName,
        isDoctor: false,
        isSuperuser: false,
      };
      firebase.firestore().collection("users").doc(user.uid).set(account);
      dispatch(registrationFinished(user));
    })
    .catch((error) => {
      dispatch(registrationError(error));
    });
};
