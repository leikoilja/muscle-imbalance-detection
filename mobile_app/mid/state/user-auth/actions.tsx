import { firebase } from "../../firebase/config";
import {
  LogoutFinishedAction,
  User,
  LoginStartAction,
  LoginFinishedAction,
  LoginErrorAction,
  LoginUserAction,
  LogoutUserAction,
  UserAuthAction,
  LogoutStartAction,
  LogoutErrorAction,
} from "../types";
import { Dispatch } from "redux";

export enum USER_AUTH_ACTION_TYPES {
  LOGIN_START = "USER_AUTH/LOGIN_START",
  LOGIN_FINISHED = "USER_AUTH/LOGIN_FINISHED",
  LOGIN_ERROR = "USER_AUTH/LOGIN_ERROR",
  LOGOUT_START = "USER_AUTH/LOGOUT_START",
  LOGOUT_FINISHED = "USER_AUTH/LOGOUT_FINISHED",
  LOGOUT_ERROR = "USER_AUTH/LOGOUT_ERROR",
}

// Login
const loginStart = (): LoginStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_START,
});

const loginFinished = (user: User): LoginFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_FINISHED,
  user,
});

const loginError = (error: string): LoginErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_ERROR,
  error,
});

// Logout
const logoutStart = (): LogoutStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_START,
});

const logoutFinished = (): LogoutFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_FINISHED,
});

const logoutError = (error: string): LogoutErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_ERROR,
  error,
});

//
// Action creators
//
export const loginUser = (
  email: string,
  password: string
): LoginUserAction => async (dispatch) => {
  dispatch(loginStart());
  try {
    const response = await firebase
      .auth()
      .signInWithEmailAndPassword(email, password);
    if (response.error) {
      throw new Error(response);
    }
    dispatch(loginFinished(response.user));
  } catch (error) {
    console.log(error);
    dispatch(loginError(error));
  }
};

export const logoutUser = (): LogoutUserAction => async (
  dispatch: Dispatch<UserAuthAction>
) => {
  dispatch(logoutStart());
  try {
    await firebase.auth().signOut();
    dispatch(logoutFinished());
  } catch (error) {
    dispatch(logoutError(error));
  }
};
