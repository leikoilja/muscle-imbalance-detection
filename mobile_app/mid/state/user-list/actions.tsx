import {
  AddUserAction,
  User,
  UpdateUserAction,
  RemoveUserAction,
} from "../types";

export enum USER_LIST_ACTION_TYPES {
  ADD_USER = "USER_LIST/ADD_USER",
  REMOVE_USER = "USER_LIST/REMOVE_USER",
  UPDATE_USER = "USER_LIST/UPDATE_USER",
}

export const addUser = (
  name: string,
  surname: string,
  age: number
): AddUserAction => ({
  type: USER_LIST_ACTION_TYPES.ADD_USER,
  userData: {
    name,
    surname,
    age,
  },
});

export const updateUser = (index: number, user: User): UpdateUserAction => ({
  type: USER_LIST_ACTION_TYPES.UPDATE_USER,
  index: index,
  userData: user,
});

export const removeUser = (index: number): RemoveUserAction => ({
  type: USER_LIST_ACTION_TYPES.REMOVE_USER,
  index: index,
});
