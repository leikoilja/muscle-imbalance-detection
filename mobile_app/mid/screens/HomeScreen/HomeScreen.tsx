import * as React from "react";

import { Text, View, TouchableOpacity } from "react-native";
import styles from "./styles";
import { connect } from "react-redux";
import { logoutUser } from "../../state/user-auth/actions";

class HomeScreen extends React.Component {
  onLogout = async () => {
    const { navigation, logoutUser } = this.props;
    const response = await logoutUser();
  };

  render() {
    const { user } = this.props.auth;

    return (
      <View style={styles.container}>
        <Text>Welcome, {user.email} </Text>
        <TouchableOpacity style={styles.button} onPress={this.onLogout}>
          <Text style={styles.buttonTitle}>Log out</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const mapStateToProps = (state) => ({
  auth: state.userAuth,
});

const mapDispatchToProps = {
  logoutUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
