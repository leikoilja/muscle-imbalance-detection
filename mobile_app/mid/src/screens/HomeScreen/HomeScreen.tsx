import * as React from "react";

import { Text, View, TouchableOpacity } from "react-native";
import styles from "./styles";
import { connect } from "react-redux";
import { logoutUser } from "../../state/user-auth/actions";
import Button from "../../components/Button";

class HomeScreen extends React.Component {
  onLogout = async () => {
    const { logoutUser } = this.props;
    await logoutUser();
  };

  onSettings = () => {
    const { navigation } = this.props;
    navigation.navigate("Settings");
  };

  render() {
    const { user } = this.props.auth;

    return (
      <View style={styles.container}>
        <Text testID="welcome-text">Welcome, {user.fullName} </Text>
        {user.isDoctor && <Text testID="doctor-info">You are doctor!</Text>}
        <Button
          style={{ width: "50%" }}
          onPress={this.onLogout}
          text="Log out"
        />
        <Text>
          Looks like you have no Bluetooth module setup, please go to 'My
          Profile' and navigate to 'Settings' to do so!
        </Text>
        <Button
          style={{ width: "50%" }}
          onPress={this.onSettings}
          text="Settings"
        />
      </View>
    );
  }
}

const mapStateToProps = (state) => ({
  auth: state.userAuth,
});

const mapDispatchToProps = {
  logoutUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
