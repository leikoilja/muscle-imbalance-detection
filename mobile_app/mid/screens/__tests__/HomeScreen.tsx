import React from "react";
import HomeScreen from "../HomeScreen/HomeScreen";
import { render, fireEvent } from "@testing-library/react-native";
import { setMockStore } from "../../utils/test_utils";
import { logoutUser } from "../../state/user-auth/actions";

describe("HomeScreen", () => {
  let store;
  let component;

  beforeEach(() => {
    [store, component] = setMockStore(
      { userAuth: { user: { fullName: "Full Name" } } },
      <HomeScreen />
    );
    store.dispatch = jest.fn();
  });

  it("shows welcome screen for user", async () => {
    const { getByTestId, toJSON } = render(component);
    expect(getByTestId("welcome-text")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("shows extra content for doctor", async () => {
    [store, component] = setMockStore(
      {
        userAuth: { user: { fullName: "Full Name", isDoctor: true } },
      },
      <HomeScreen />
    );
    const { getByTestId, toJSON } = render(component);
    expect(getByTestId("doctor-info")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("loggs-out when Logout button is pressed", async () => {
    const { getByText } = render(component);
    const button = getByText("Log out");
    fireEvent(button, "press");
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    // TODO: Check that logoutUser dispatch has been called
  });
});
