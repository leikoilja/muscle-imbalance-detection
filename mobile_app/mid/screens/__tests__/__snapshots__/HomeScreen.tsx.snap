// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen shows extra content for doctor 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="welcome-text"
  >
    Welcome, 
    Full Name
     
  </Text>
  <Text
    testID="doctor-info"
  >
    You are doctor!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Log out
    </Text>
  </View>
</View>
`;

exports[`HomeScreen shows welcome screen for user 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="welcome-text"
  >
    Welcome, 
    Full Name
     
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Log out
    </Text>
  </View>
</View>
`;
